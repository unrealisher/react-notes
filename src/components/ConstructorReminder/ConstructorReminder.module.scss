.text {
  margin-right: 5px;
}

.input {
  padding: 2px 10px;
  border-radius: 15px;
}
