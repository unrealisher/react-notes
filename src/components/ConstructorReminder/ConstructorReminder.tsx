import React, { useRef } from "react";
import moment from "moment";

import styles from "./ConstructorReminder.module.scss";

interface IProps {
  reminder: number;
  setReminder: Function;
}

export const ConstructorReminder = (props: IProps): JSX.Element => {
  const { reminder, setReminder } = props;
  const inputRef = useRef<HTMLInputElement>(null);
  return (
    <label>
      <span className={styles.text}>Напоминание:</span>
      <input
        className={styles.input}
        type="date"
        value={
          reminder !== 0 ? moment(reminder).format("YYYY-MM-DD") : undefined
        }
        ref={inputRef}
        onChange={() => {
          if (inputRef.current && inputRef.current.valueAsDate) {
            setReminder(inputRef.current.valueAsDate.getTime());
          } else {
            setReminder(0);
          }
        }}
      />
    </label>
  );
};

export default ConstructorReminder;
