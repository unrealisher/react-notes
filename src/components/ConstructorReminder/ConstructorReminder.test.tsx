import React from "react";
import { shallow } from "enzyme";
import { ConstructorReminder } from "./ConstructorReminder";

describe("ConstructorReminder", () => {
  const reminder = 100;
  const setReminder = (): void => {};
  it("correct render", () => {
    const component = shallow(
      <ConstructorReminder reminder={reminder} setReminder={setReminder} />
    );
    expect(component).toMatchSnapshot();
  });
});
