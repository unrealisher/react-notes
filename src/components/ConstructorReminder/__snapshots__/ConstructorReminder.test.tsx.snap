// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConstructorReminder correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConstructorReminder
    reminder={100}
    setReminder={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="text"
        >
          Напоминание:
        </span>,
        <input
          className="input"
          onChange={[Function]}
          type="date"
          value="1970-01-01"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Напоминание:",
          "className": "text",
        },
        "ref": null,
        "rendered": "Напоминание:",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input",
          "onChange": [Function],
          "type": "date",
          "value": "1970-01-01",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="text"
          >
            Напоминание:
          </span>,
          <input
            className="input"
            onChange={[Function]}
            type="date"
            value="1970-01-01"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Напоминание:",
            "className": "text",
          },
          "ref": null,
          "rendered": "Напоминание:",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input",
            "onChange": [Function],
            "type": "date",
            "value": "1970-01-01",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
